internal/keycloakb/configdbmodule.go

```diff
@@ -0,0 +1,74 @@
+package keycloakb
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+
+	"github.com/cloudtrust/keycloak-bridge/internal/dto"
+)
+
+const (
+	updateConfigStmt = `INSERT INTO realm_configuration (realm_id, configuration) 
+	  VALUES (?, ?) 
+	  ON DUPLICATE KEY UPDATE configuration = ?;`
+	selectConfigStmt = `SELECT configuration FROM realm_configuration WHERE (realm_id = ?)`
+)
+
+// DBConfiguration interface
+type DBConfiguration interface {
+	Exec(query string, args ...interface{}) (sql.Result, error)
+	QueryRow(query string, args ...interface{}) *sql.Row
+}
+
+type configurationDBModule struct {
+	db DBConfiguration
+}
+
+// MissingRealmConfigurationErr is the error thrown if no configuration is found in DB
+type MissingRealmConfigurationErr struct {
+	realmID string
+}
+
+func (e MissingRealmConfigurationErr) Error() string {
+	return "RealmConfiguration is not configured for " + e.realmID
+}
+
+// NewConfigurationDBModule returns a ConfigurationDB module.
+func NewConfigurationDBModule(db DBConfiguration) *configurationDBModule {
+	return &configurationDBModule{
+		db: db,
+	}
+}
+
+func (c *configurationDBModule) StoreOrUpdate(context context.Context, realmID string, config dto.RealmConfiguration) error {
+	// transform customConfig object into JSON string
+	configJSON, err := json.Marshal(config)
+	if err != nil {
+		return err
```

Comment:
Shall we use specific error codes ?

Reply:
If we have such an issue, it seems to be an internal server error (500) so I don't see any benefit of using specific error codes. WDYT?

Reply:
I agree !
(I read too fast... I thought it was an error related to a DB call)